Upstream has not yet decided on this
https://gitlab.torproject.org/tpo/core/tor/-/merge_requests/273/diffs
so we only patch the test.
--- a/src/test/test_util.c
+++ b/src/test/test_util.c
@@ -4562,15 +4562,8 @@
   TEST("file1");
   EXPECT(results_test9);
 
-#if defined(__APPLE__) || defined(__darwin__) || \
-  defined(__FreeBSD__) || defined(__NetBSD__) || defined(OpenBSD)
   TEST("file1"PATH_SEPARATOR);
   EXPECT_EMPTY();
-#else
-  const char *results_test10[] = {"file1"};
-  TEST("file1"PATH_SEPARATOR);
-  EXPECT(results_test10);
-#endif
 
   // test path separator at end - with wildcards and linux path separator
   const char *results_test11[] = {"dir1", "dir2", "forbidden"};
